### PR TITLE
HostConnectionPool: log whether port-based shard awareness is used

### LINKED ISSUE #112: Log whether the driver is using port based shard awareness or not and why

### PR DESCRIPTION
**Pull request by haaawk**
If that's not the case log the reason why.

Fixes #112

Signed-off-by: Piotr Jastrzebski <piotr@scylladb.com>

### CODE REVIEW

File: driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java [hunk 1]
```diff
@@ -158,24 +158,51 @@ public Connection addConnectionToClose(int shardId, Connection c) {
...
       return false;
     }
 
     boolean isSSLUsed = null != manager.configuration().getProtocolOptions().getSSLOptions();
     if (shardingInfo.getShardAwarePort(isSSLUsed) == 0) {
+      if (logResult) {
+        logger.warn(
+            "Not using advanced port-based shard awareness with "
+                + host
+                + " because it's impossible to get the outgoing port");
```
Comment (Lorak-mmk) [inline driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java]: I think this message may be wrong. The check isn't about allocating local, outgoing port (it happens elsewhere), it's about whether the server sent us it's shard-aware port number.

**Review by Lorak-mmk — CHANGES_REQUESTED** [review]

File: driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java (hunk 1 above)
Comment (haaawk) [reply driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java]: Fair point. Thanks. Fixed.

**Review by Lorak-mmk — APPROVED** [review]

File: driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java [hunk 2]
```diff
@@ -158,24 +158,51 @@ public Connection addConnectionToClose(int shardId, Connection c) {
...
+        logger.warn(
+            "Not using advanced port-based shard awareness with "
+                + host
+                + " because of a previous error");
+      }
       return false;
     }
 
+    if (logResult) {
+      logger.info("Using advanced port-based shard awareness with " + host);
```
Comment (avelanarius) [inline driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java:204]:
```suggestion
      logger.info("Using advanced port-based shard awareness with {}", host);
```

(similar fix also in other `logger` calls)

**Review by avelanarius — APPROVED** [review]

**Comment by avelanarius** [comment]
Merged via 12a712ff5187b5a2f2ec0f32c186e4cae0954de0

**Pull request closed without merging**